core/src/main/java/de/danoeh/antennapod/core/util/ChapterUtils.java

```diff
@@ -123,6 +135,23 @@ private static CountingInputStream openStream(Playable playable, Context context
         }
     }
 
+    public static List<Chapter> loadChaptersFromUrl(String url) {
+        List<Chapter> chapters;
+        Request request = new Request.Builder().url(url).build();
+        try {
+            Response response = AntennapodHttpClient.getHttpClient().newCall(request).execute();
+            if (response.body() == null) {
+                return null;
+            }
+            chapters = PodcastIndexChapter.parseChapters(response.body().toString());
+            return chapters;
+        } catch (IOException e) {
+            Log.d(TAG, "error loading data from " + url);
+        } finally {
+            return null;
```

Comment:
Returning in a `finally` block is a bit weird because the `finally` block is executed after the `try` block and therefore after the actual return above. Please move the return out of the block.